### PR TITLE
Replace usage of eventing v1beta1 APIs with v1

### PR DESCRIPTION
**Pull request by grantr**
The v1beta1 versions of Broker and Trigger have been removed from upstream. The v1 version has an identical schema, so replacing is just import juggling.

This should unblock the auto dependency update.

## Proposed Changes

- 🗑️ Remove support for Broker and Trigger v1beta1

**Release Note**

```release-note
## Action Required
The v1beta1 versions of Broker and Trigger are no longer supported. Please switch to the v1 versions of these resources (schema is identical).
```

### CODE REVIEW

**Comment by grantr** [comment]
Seems like we inherited the change to remove self-approval from Knative test-infra. This change is likely to be reverted in Knative, so we probably don't need to do anything to revert it ourselves.

File: test/lib/helpers/broker_event_transformation_test_with_source.go [hunk 1]
```diff
@@ -502,26 +502,26 @@ func createFirstNErrsReceiver(client *lib.Client, firstNErrs int) string {
...
-	triggerFilter eventingtestresources.TriggerOptionV1Beta1) *eventingv1beta1.Trigger {
+	triggerFilter eventingtestresources.TriggerOptionV1Beta1) *eventingv1.Trigger {
 	client.T.Helper()
 	respTriggerName := "resp-broker-" + brokerName
 	return client.Core.CreateTriggerOrFailV1Beta1(
 		respTriggerName,
-		eventingtestresources.WithBrokerV1Beta1(brokerName),
+		eventingtestresources.WithBrokerV1(brokerName),
 		triggerFilter,
 		eventingtestresources.WithSubscriberServiceRefForTriggerV1Beta1(targetName),
```
Comment (eclipselu) [inline test/lib/helpers/broker_event_transformation_test_with_source.go:526]: This needs to be changed to V1 versions as well.

File: test/lib/helpers/broker_event_transformation_test_with_source.go [hunk 2]
```diff
@@ -502,26 +502,26 @@ func createFirstNErrsReceiver(client *lib.Client, firstNErrs int) string {
...
+	triggerFilter eventingtestresources.TriggerOptionV1Beta1) *eventingv1.Trigger {
 	client.T.Helper()
 	// Please refer to the graph in the file to check what sample trigger is used for.
 	triggerName := "trigger-broker-" + brokerName
 	return client.Core.CreateTriggerOrFailV1Beta1(
 		triggerName,
-		eventingtestresources.WithBrokerV1Beta1(brokerName),
+		eventingtestresources.WithBrokerV1(brokerName),
 		triggerFilter,
 		eventingtestresources.WithSubscriberServiceRefForTriggerV1Beta1(kserviceName),
```
Comment (eclipselu) [inline test/lib/helpers/broker_event_transformation_test_with_source.go:513]: This needs to be changed to [client.Core.CreateTriggerOrFail](https://github.com/knative/eventing/blob/main/test/lib/creation.go#L251), and WithSubscriberServiceRefForTriggerV1

**Review by eclipselu — CHANGES_REQUESTED** [review]

File: test/lib/helpers/broker_event_transformation_test_with_source.go [hunk 3]
```diff
@@ -502,26 +502,26 @@ func createFirstNErrsReceiver(client *lib.Client, firstNErrs int) string {
 
 func createTriggerWithKServiceSubscriber(client *lib.Client,
 	brokerName, kserviceName string,
-	triggerFilter eventingtestresources.TriggerOptionV1Beta1) *eventingv1beta1.Trigger {
+	triggerFilter eventingtestresources.TriggerOptionV1Beta1) *eventingv1.Trigger {
 	client.T.Helper()
 	// Please refer to the graph in the file to check what sample trigger is used for.
 	triggerName := "trigger-broker-" + brokerName
 	return client.Core.CreateTriggerOrFailV1Beta1(
```
Comment (eclipselu) [inline test/lib/helpers/broker_event_transformation_test_with_source.go:509]:
```suggestion
	return client.Core.CreateTriggerOrFail(
```

File: test/lib/helpers/broker_event_transformation_test_with_source.go (hunk 2 above)
Comment (eclipselu) [inline test/lib/helpers/broker_event_transformation_test_with_source.go:513]:
```suggestion
		eventingtestresources.WithSubscriberServiceRefForTriggerV1(kserviceName),
```

File: test/lib/helpers/broker_event_transformation_test_with_source.go (hunk 1 above)
Comment (eclipselu) [inline test/lib/helpers/broker_event_transformation_test_with_source.go:526]:
```suggestion
		eventingtestresources.WithSubscriberServiceRefForTriggerV1(targetName),
```

File: test/lib/helpers/broker_event_transformation_test_with_source.go [hunk 4]
```diff
@@ -502,26 +502,26 @@ func createFirstNErrsReceiver(client *lib.Client, firstNErrs int) string {
...
 	)
 }
 
 func createTriggerWithTargetServiceSubscriber(client *lib.Client,
 	brokerName, targetName string,
-	triggerFilter eventingtestresources.TriggerOptionV1Beta1) *eventingv1beta1.Trigger {
+	triggerFilter eventingtestresources.TriggerOptionV1Beta1) *eventingv1.Trigger {
 	client.T.Helper()
 	respTriggerName := "resp-broker-" + brokerName
 	return client.Core.CreateTriggerOrFailV1Beta1(
```
Comment (eclipselu) [inline test/lib/helpers/broker_event_transformation_test_with_source.go:522]:
```suggestion
	return client.Core.CreateTriggerOrFail(
```

**Comment by danyinggu** [comment]
It seems like we need to run update-codegen.sh again.

**Comment by grantr** [comment]
I'm unable to get Prow's update-codegen.sh output locally. When I run it I don't get a diff. Maybe I have an environment difference?

**Pull request closed without merging**